lib/src/facts/collection.cc

```diff
@@ -383,12 +383,17 @@ namespace facter { namespace facts {
         return current;
     }
 
-    value const* collection::lookup(value const* value, string const& name)
+    value const* collection::lookup(value const* value, string const& name, bool detailed_exit_codes)
```

Comment:
This could obviously be simplified by making the flag part of the `collection` object. If these were all public functions, I'd lean towards adding it on construction; since they're all implementation and only affect one public function, I like keeping it restricted to that function rather than a property of the `collection`.